backend/src/main.rs

```diff
@@ -4,22 +4,48 @@ extern crate log;
 use std::net::Ipv4Addr;
 
 use actix::prelude::*;
-use actix_web::{web, App, HttpRequest, HttpResponse, HttpServer, Error};
-use actix_web_actors::ws;
 use actix_http::ws::Codec;
+use actix_web::{web, App, Error, HttpRequest, HttpResponse, HttpServer};
+use actix_web_actors::ws;
+use clap::Clap;
 
-mod types;
 mod aggregator;
 mod chain;
-mod node;
 mod feed;
+mod node;
+mod types;
 mod util;
 
-use node::connector::NodeConnector;
+use aggregator::{Aggregator, GetHealth, GetNetworkState};
 use feed::connector::FeedConnector;
-use aggregator::{Aggregator, GetNetworkState, GetHealth};
-use util::{Locator, LocatorFactory};
+use node::connector::NodeConnector;
 use types::NodeId;
+use util::{Locator, LocatorFactory};
+
+const VERSION: &'static str = env!("CARGO_PKG_VERSION");
+const AUTHORS: &'static str = env!("CARGO_PKG_AUTHORS");
+const NAME: &'static str = "Substrate Telemetry Backend";
+const ABOUT: &'static str = "This is the Telemetry Backend that injects and provide the data sent by Substrate/Polkadot nodes";
+
+#[derive(Clap)]
+#[clap(name = NAME, version = VERSION, author = AUTHORS, about = ABOUT)]
+struct Opts {
+    #[clap(
+        short = "p",
+        long = "port",
+        default_value = "8000",
+        help = "This is the port Telemetry is listening to."
+    )]
+    port: u16,
+
+    #[clap(
+        short = "b",
+        long = "bind",
+        default_value = "127.0.0.1",
+        help = "This is the address of the network interface Telemetry is listening to. This is restricted localhost by default and should be fine for most use cases. If you are using Telemetry in a container, you likely want to set this to '0.0.0.0'"
+    )]
+    bind: String,
```

Comment:
Could try to make this a [`std::net::IpAddr`](https://doc.rust-lang.org/std/net/enum.IpAddr.html), so that if there is any user error in the CLI, the parsing fails at clap level, instead of when trying to spin the server.

Reply:
💯Let me improve that

Reply:
Fixed using SocketAddr

Reply:
`SocketAddr` includes port, so that would make port redundant.